tests/unit/test_stats_reporter.py

```diff
@@ -279,3 +341,108 @@ def test_draw_histogram_invalid_input():
     # test#3 - Invalid hist_scale_factor value
     with pytest.raises(ValueError):
         _ = draw_histogram([100, 200, 300], bins=5, hist_scale_factor=0)
+
+
+def test_stats_output(fake_stats):
+    reporter = StatsReporter(fake_stats, 5)
+    with patch("builtins.print") as mocked_print:
+        with patch("rich.print", print):
+            reporter.render()
+    expected = (
+        "📏 [bold]Total allocations:[/]\n"
+        "\t20\n"
+        "\n"
+        "📦 [bold]Total memory allocated:[/]\n"
+        "\t3.187MB\n"
+        "\n"
+        "📊 [bold]Histogram of allocation size:[/]\n"
+        "\tmin: 2.441KB\n"
+        "\t----------------------------------------\n"
+        "\t< 4.628KB  : 1 ▇▇▇▇▇\n"
+        "\t< 8.775KB  : 0 \n"
+        "\t< 16.637KB : 3 ▇▇▇▇▇▇▇▇▇▇▇▇▇\n"
+        "\t< 31.542KB : 0 \n"
+        "\t< 59.802KB : 1 ▇▇▇▇▇\n"
+        "\t< 113.378KB: 1 ▇▇▇▇▇\n"
+        "\t< 214.954KB: 6 ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇\n"
+        "\t< 407.531KB: 0 \n"
+        "\t< 772.638KB: 0 \n"
+        "\t<=1.431MB  : 2 ▇▇▇▇▇▇▇▇▇\n"
+        "\t----------------------------------------\n"
+        "\tmax: 1.431MB\n"
+        "\n"
+        "📂 [bold]Allocator type distribution:[/]\n"
+        "\t 1: 1013\n"
+        "\t 4: 797\n"
+        "\t 3: 152\n"
+        "\t 10: 4\n"
```

Comment:
Once `fake_stats` is fixed to pass strings, this will need to be:

```suggestion
        "\t MALLOC: 1013\n"
        "\t REALLOC: 797\n"
        "\t CALLOC: 152\n"
        "\t MMAP: 4\n"
```